src/couch_httpd.erl

```diff
@@ -1127,11 +1127,10 @@ check_for_last(#mp{buffer=Buffer, data_fun=DataFun}=Mp) ->
                 data_fun = DataFun2})
     end.
 
+validate_bind_address(any) -> any;
+validate_bind_address("any") -> any;
```

Comment:
I don't think the validate_bind_address function should work this way, it should _only_ validate. It now returns 'ok' or 'any' or throws, which is quite confusing. The intention is correct though, to allow config to hold "any" and have that be the atom any. Have another go :)